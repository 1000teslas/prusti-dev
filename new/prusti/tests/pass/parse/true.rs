// compile-flags: -Zprint-desugared-specs

#![feature(register_tool)]
#![register_tool(prusti)]
#![feature(stmt_expr_attributes)]
#![feature(custom_inner_attributes)]

#[prusti::requires(true)]
fn test1() {}

#[prusti::ensures(true)]
fn test2() {}

fn test3() {
    for _ in 0..2 {
        #![prusti::invariant(true)]
    }
}

#[prusti::requires(true)]
#[prusti::ensures(true)]
fn test4() {
    for _ in 0..2 {
        #![prusti::invariant(true)]
    }
}

fn main() {}
