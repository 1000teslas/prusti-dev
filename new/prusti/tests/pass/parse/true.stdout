// compile-flags: -Zprint-desugared-specs

#![feature(register_tool)]
#![register_tool(prusti)]
#![feature(stmt_expr_attributes)]
#![feature(custom_inner_attributes)]
#[prelude_import]
use std::prelude::v1::*;
#[macro_use]
extern crate std;

#[prusti::spec_id(101u64)]
fn test1() { }

#[prusti::spec_id(102u64)]
fn test2() { }

fn test3() {
    for _ in 0..2 {
        {

            #[prusti::invariant_spec]
            if false { }
        }
    }
}

#[prusti::spec_id(103u64)]
fn test4() {
    for _ in 0..2 {
        {

            #[prusti::invariant_spec]
            if false { }
        }
    }
}

fn main() { }
#[prusti::spec_only]
fn prusti_spec_item_103() {
    #[prusti::spec_only]
    fn prusti_spec_item_103_pre() { || -> bool { (true) }; }
    #[prusti::spec_only]
    fn prusti_spec_item_103_post(result: ()) { || -> bool { (true) }; }
}
#[prusti::spec_only]
fn prusti_spec_item_102() {
    #[prusti::spec_only]
    fn prusti_spec_item_102_pre() { }
    #[prusti::spec_only]
    fn prusti_spec_item_102_post(result: ()) { || -> bool { (true) }; }
}
#[prusti::spec_only]
fn prusti_spec_item_101() {
    #[prusti::spec_only]
    fn prusti_spec_item_101_pre() { || -> bool { (true) }; }
    #[prusti::spec_only]
    fn prusti_spec_item_101_post(result: ()) { }
}
HIR Item: 
  Not a function
HIR Item: std
  Not a function
HIR Item: test3
  Return type is unit: true
HIR Item: main
  Return type is unit: true
HIR Item: test1
  Return type is unit: true
HIR Item: test2
  Return type is unit: true
HIR Item: test4
  Return type is unit: true
HIR Item: prusti_spec_item_103
  Return type is unit: true
HIR Item: prusti_spec_item_103_pre
  Return type is unit: false
HIR Item: prusti_spec_item_103_post
  Return type is unit: false
HIR Item: prusti_spec_item_102
  Return type is unit: true
HIR Item: prusti_spec_item_102_pre
  Return type is unit: true
HIR Item: prusti_spec_item_102_post
  Return type is unit: false
HIR Item: prusti_spec_item_101
  Return type is unit: true
HIR Item: prusti_spec_item_101_pre
  Return type is unit: false
HIR Item: prusti_spec_item_101_post
  Return type is unit: true
